sklearn/linear_model/least_angle.py

```diff
@@ -1468,8 +1469,9 @@ def fit(self, X, y, copy_X=True):
             df[k] = np.sum(mask)
 
         self.alphas_ = alphas_
-        with np.errstate(divide='ignore'):
-            self.criterion_ = n_samples * np.log(mean_squared_error) + K * df
+        eps64 = np.finfo('float64').eps
+        self.criterion_ = (n_samples * mean_squared_error / (sigma2 + eps64) +
```

Comment:
also maybe document somewhere that the criterion is off by a constant and scaled by n, compared to the actual value, but that this doesn't affect relative comparisons? The [wikipedia page for AIC](https://en.wikipedia.org/wiki/Akaike_information_criterion) has a section calling this "software unreliability"...